[COPY] --- T2-COPYRIGHT-BEGIN ---
[COPY] t2/package/*/linux/hotfix-kern-cont-windtunnel.desc
[COPY] Copyright (C) 2025 The T2 SDE Project
[COPY] SPDX-License-Identifier: GPL-2.0
[COPY] --- T2-COPYRIGHT-END ---

diff --git a/drivers/macintosh/therm_windtunnel.c b/drivers/macintosh/therm_windtunnel.c
index 0b2e08a1bee0..5ffc4711847d 100644
--- a/drivers/macintosh/therm_windtunnel.c
+++ b/drivers/macintosh/therm_windtunnel.c
@@ -160,8 +160,8 @@ tune_fan( int fan_setting )
 	write_reg( x.fan, 0x20, 0, 1 );
 	print_temp("CPU-temp: ", x.temp );
 	if( x.casetemp )
-		print_temp(", Case: ", x.casetemp );
-	printk(",  Fan: %d (tuned %+d)\n", 11-fan_setting, x.fan_level-fan_setting );
+		print_temp(KERN_CONT ", Case: ", x.casetemp );
+	printk(KERN_CONT ",  Fan: %d (tuned %+d)\n", 11-fan_setting, x.fan_level-fan_setting );
 
 	x.fan_level = fan_setting;
 }
@@ -182,8 +182,8 @@ poll_temp( void )
 
 	if( LOG_TEMP && x.temp != temp ) {
 		print_temp("CPU-temp: ", temp );
-		print_temp(", Case: ", casetemp );
-		printk(",  Fan: %d\n", 11-x.fan_level );
+		print_temp(KERN_CONT ", Case: ", casetemp );
+		printk(KERN_CONT ",  Fan: %d\n", 11-x.fan_level );
 	}
 	x.temp = temp;
 	x.casetemp = casetemp;
@@ -244,8 +244,8 @@ setup_hardware( void )
 		write_reg( x.thermostat, 3, x.overheat_temp, 2 );
 
 		print_temp("Reducing overheating limit to ", x.overheat_temp );
-		print_temp(" (Hyst: ", x.overheat_hyst );
-		printk(")\n");
+		print_temp(KERN_CONT " (Hyst: ", x.overheat_hyst );
+		printk(KERN_CONT ")\n");
 	}
 
 	/* set an initial fan setting */
@@ -390,10 +390,10 @@ attach_thermostat( struct i2c_client *cl )
 		goto out;
 
 	printk("DS1775 digital thermometer [@%02x]\n", cl->addr );
-	print_temp("Temp: ", temp );
-	print_temp("  Hyst: ", hyst_temp );
-	print_temp("  OS: ", os_temp );
-	printk("\n");
+	print_temp(KERN_CONT "Temp: ", temp );
+	print_temp(KERN_CONT "  Hyst: ", hyst_temp );
+	print_temp(KERN_CONT "  OS: ", os_temp );
+	printk(KERN_CONT "\n");
 
 	x.temp = temp;
 	x.overheat_temp = os_temp;
